src/home/user-record-util.c

```diff
@@ -808,7 +809,7 @@ int user_record_make_hashed_password(UserRecord *h, char **secret, bool extend)
         STRV_FOREACH(i, secret) {
                 _cleanup_free_ char *hashed = NULL;
 
-                r = hash_password(*i, &hashed);
+                r = hash_password_full(*i, &cd_data, &cd_size, &hashed);
```

Comment:
i really wouldn't expose the reuse of the structure here at all. it's not that we hash a million paswords. We generally hash one, and in some exceptional cases maybe 2 or 3, but paying a malloc() for each should not matter.

hence, let's hide the whole allocation/deallocation logic inside of hash_password() please